packages/playwright-core/src/server/socksClientCertificatesInterceptor.ts

```diff
@@ -98,11 +98,14 @@ class SocksProxyConnection {
     this.internalTLS = internalTLS;
     internalTLS.on('close', () => this.socksProxy._socksProxy.sendSocketEnd({ uid: this.uid }));
 
-    const targetTLS = tls.connect({
+    const tlsOptions: tls.ConnectionOptions = {
       socket: this.target,
-      rejectUnauthorized: this.socksProxy.contextOptions.ignoreHTTPSErrors === true ? false : true,
-      ...clientCertificatesToTLSOptions(this.socksProxy.contextOptions.clientCertificates, `https://${this.host}:${this.port}/`),
-    });
+      rejectUnauthorized: this.socksProxy.ignoreHTTPSErrors === true ? false : true,
+      ...clientCertificatesToTLSOptions(this.socksProxy.clientCertificates, `https://${this.host}:${this.port}/`),
+    };
+    if (process.env.PWTEST_UNSUPPORTED_CUSTOM_CA && isUnderTest())
```

Comment:
we have a similar hack here https://github.com/microsoft/playwright/blob/6dbc7b54e8ea62c1f31c43aa8dad7e079e136003/packages/playwright-core/src/server/fetch.ts#L199 to pretend to have a real-world site.